### ISSUE TITLE
Mirroring displays: hang

### ISSUE THREAD

**Issue by CottonEaster** [linuxmint/Cinnamon#7543]
```
 * Cinnamon version (cinnamon --version): 3.6.7
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 18.3
 * Graphics hardware *and* driver used: not relevant, I think
 * 32 or 64 bit: 64
 ```

**Issue**

Upon connecting my Mint laptop to an external monitor, and mirroring the displays, Mint hung. (And, upon reboot via a tty - which was all that was possible - two of my icons were misplaced and initially the screen resolution on the laptop was wrong.)

**Steps to reproduce**

1. Connect laptop (mine: an X1 Carbon) to an external monitor (mine: a Dell U2412M, connected via a HDMI-to-VGA adaptor and a VGA cable).
2. Tell Cinnamon to mirror the displays.
3. At the 'keep or reject' dialogue, choose 'reject'.

**Expected behaviour**

3 should not hang the computer and the attendant problems should not occur.

**Other information**

I attach a screenshot, taken by a phone camera, of the Mint laptop's screen when it hung.

![image](https://user-images.githubusercontent.com/9998846/39608433-63bd4e46-4f39-11e8-800a-677be1e8f375.png)

**Comment by leigh123linux** [comment]
Plug in the external monitor, don't try to set up and post the output for this command

```
xrandr
```

**Comment by JosephMcc** [comment]
I haven't tried to reproduce this yet but how can figure the video hardware and driver aren't relevant when you are talking about display issues? The template is there for a reason please use it.

**Comment by CottonEaster** [comment]
> how can figure the video hardware and driver aren't relevant when you are talking about display issues? 

Good point. Sorry. I suppose I had got use to that information being irrlevant for some of the bugs I've filed.

- Intel Open Source Technology Center
- Mesa DRI Intel Kabylake GT1.5
- 3.0 Mesa 17.2.8
- Direct Rendering: yes
- Resolution: 2560x1440
- DPI: double

```
$ xrandr
Screen 0: minimum 8 x 8, current 4480 x 1440, maximum 32767 x 32767
eDP1 connected primary 2560x1440+0+0 (normal left inverted right x axis y axis) 309mm x 174mm
   2560x1440     60.01*+
   1920x1440     60.00  
   1856x1392     60.01  
   1792x1344     60.01  
   2048x1152     60.00  
   1920x1200     59.95  
   1920x1080     60.00    59.93  
   1600x1200     60.00  
   1680x1050     59.95    59.88  
   1600x1024     60.17  
   1400x1050     59.98  
   1600x900      60.00  
   1280x1024     60.02  
   1440x900      59.89  
   1280x960      60.00  
   1368x768      60.00  
   1360x768      59.80    59.96  
   1152x864      60.00  
   1280x720      60.00  
   1024x768      60.00  
   1024x576      60.00  
   960x540       60.00  
   800x600       60.32    56.25  
   864x486       60.00  
   640x480       59.94  
   720x405       60.00  
   640x360       60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
HDMI1 connected 1920x1200+2560+0 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
   1920x1080     60.00  
   1600x1200     60.00  
   1680x1050     59.88  
   1280x1024     60.02  
   1280x960      60.00  
   1024x768      60.00  
   800x600       60.32  
   640x480       59.94  
   720x400       70.08  
HDMI2 disconnected (normal left inverted right x axis y axis)
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
```

**Comment by CottonEaster** [comment]
It might be worth adding that, having forgotten that the external monitor was plugged in (but not manually set up in any way), I used alt-tab and again got a hang. Interestingly, removing the HMDI cable unfroze the computer.

**Comment by mtwebster** [comment]
apt-get remove xserver-xorg-video-intel

**Comment by CottonEaster** [comment]
I would rather keep `xserver-xorg-video-intel` because, though it is somewhat depreciated, without it, [my battery power does not display properly](https://github.com/linuxmint/Cinnamon/issues/5831) (please see the end of that thread). Or is this removal a temporary one for testing?

**Comment by CottonEaster** [comment]
Under Mint 19, and with `xserver-xorg-video-intel` removed, there is no longer any hang (though icons do get re-arranged when I return, on my Linux laptop, to an unmirrored display - even if I disable mirroring and reset the resolution in a single action; in short, whatever I do, icons get re-arranged after mirroring).

**Comment by CottonEaster** [comment]
_Since the original problem here is solved by doing something that is in the Mint 18.3 release notes_ - but **not**, I believe in _later_ release notes - I will close this issue.

So, to be clear - connecting an external monitor can still cause hangs on the latest Mint, notably when alt-tabbing with 'coverflow (3d); but removing `xserver-xorg-video-intel` fixes the problem, though the Mint 19 Cinnamon and 19.1 Cinnamon release notes do not mention this.

Yet, I am still having problems with icons rearranging themselves; I will open a separate report for that.
